### PR TITLE
fix: apply farm ID and queue ID settings correctly

### LINKED ISSUE #117: Bug: login callback does not validate farm or queue values before calling Deadline APIs
### Expected Behaviour

`get_farm` or `get_queue` is called with non-empty values. 

### Current Behaviour

`get_farm` or `get_queue` is called with empty values on first load leading to undefined behavior. 

### Reproduction Steps

Remove all cached deadline settings,  then login to the submitter. 

### Code Snippet

https://github.com/aws-deadline/deadline-cloud-for-houdini/blob/mainline/src/deadline/houdini_submitter/python/deadline_cloud_for_houdini/submitter.py#L701-L707:

```
def login_callback(kwargs):
    node = kwargs["node"]
    node.parm("farm").set("<refreshing>")
    node.parm("queue").set("<refreshing>")
    DeadlineLoginDialog.login(parent=hou.qt.mainWindow())
    deadline = api.get_boto3_client("deadline")
    farm_id = get_setting("defaults.farm_id")
    farm_response = deadline.get_farm(farmId=farm_id)
    node.parm("farm").set(farm_response["displayName"])
    queue_id = get_setting("defaults.queue_id")
    queue_response = deadline.get_queue(farmId=farm_id, queueId=queue_id)
    node.parm("queue").set(queue_response["displayName"])
    update_queue_parameters(farm_id, queue_id, node)
```

### PR DESCRIPTION
**Pull request by joel-wong-aws**
### What was the problem/requirement? (What/Why)
The error message "AccessDeniedException" was appearing during login, settings, and creation flows with no actionable error message.

Resolves https://github.com/aws-deadline/deadline-cloud-for-houdini/issues/117

### What was the solution? (How)
Updated the flows to check whether the farm and queue IDs are set before calling Deadline Cloud APIs. Use a shared function for consistent behaviour.

Now, customers will only see error messages if they attempt to submit without a farm and queue ID set. The ROP creation, login, and settings flows will not show an error message if the farm and queue ID are not set.

### What is the impact of this change?
Avoids customer confusion from access denied exceptions

### How was this change tested?
1. Unset the farm ID and queue ID using `deadline config set defaults.farm_id ""` and `deadline config set defaults.queue_id`
2. Created a simple geometry, camera, and light scene with a Mantra ROP
3. Created a Deadline Cloud Node 
Before: access denied error appears
After: no error
4. Clicked login
Before: access denied error appears, even though login succeeded
After: no error
3. Opened the settings menu, then clicked okay without setting the farm and queue
Before: access denied error appears, even though there was no action by the user
After: no error
5. Submitted the job. An error message that the farm ID was not set appeared
6. Set the farm ID using `deadline config set defaults.farm_id <farm ID>`
7. Created a Deadline Cloud Node
Before: access denied error appears
After: no error
8. Clicked login
Before: access denied error appears, even though login succeeded
After: no error
9. Opened the settings menu, then clicked okay without setting the farm and queue
Before: access denied error appears, even though there was no action by the user
After: no error
10. Submitted the job. An error message that the queue ID was not set appeared
11. Set the queue ID using `deadline config set defaults.queue_id <queue ID>`
12. Created a Deadline Cloud node, clicked login, opened settings, and submitted a job

### Was this change documented?
No, not required

### Is this a breaking change?
No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

**Review by crowecawcaw — APPROVED** [review]

File: src/deadline/houdini_submitter/otls/deadline_cloud.hda/Driver_1deadline__cloud/OnCreated [hunk 1]
```diff
@@ -1,16 +1,7 @@
...
 
 try:
     node = kwargs['node']
-    deadline = get_boto3_client('deadline')
-    farm_id = get_setting('defaults.farm_id')
-    farm_response = deadline.get_farm(farmId=farm_id)
-    node.parm('farm').set(farm_response['displayName'])
-    queue_id = get_setting('defaults.queue_id')
-    queue_response = deadline.get_queue(farmId=farm_id, queueId=queue_id)
-    node.parm('queue').set(queue_response['displayName'])
```
Comment (epmog) [inline src/deadline/houdini_submitter/otls/deadline_cloud.hda/Driver_1deadline__cloud/OnCreated:13]: The result of this change means that on node creation we won't populate the farm and queue, even though they may be set in the deadline config already (managed outside of houdini)?

Can we add that back in here? Is it handled elsewhere? Or does this present other issues?

**Review by epmog — COMMENTED** [review]
Great clean-ups! I just had the one question around the OnCreated node event

File: src/deadline/houdini_submitter/otls/deadline_cloud.hda/Driver_1deadline__cloud/OnCreated (hunk 1 above)
Comment (joel-wong-aws) [reply src/deadline/houdini_submitter/otls/deadline_cloud.hda/Driver_1deadline__cloud/OnCreated:13]: This logic is now included in the `update_queue_parameters_callback` function!

Comment (epmog) [reply src/deadline/houdini_submitter/otls/deadline_cloud.hda/Driver_1deadline__cloud/OnCreated:13]: So it is! Turns out `update_queue_parameters_callback` != `update_queue_parameters`. Thanks!

**Review by epmog — APPROVED** [review]

**Pull request merged**